### ISSUE TITLE
Missing Twitter keys error is confusing

### ISSUE THREAD

**Issue by remagio** [DocNow/twarc#150]
I'm having issues in using replies and --recursive, --recursive result is a mismatch argument.
Just a sample without using the recursive argument (same with any ID):
```
twarc replies 833504567644024832 --log test.log
```
result:
```
errors
Traceback (most recent call last):
  File "/usr/local/bin/twarc", line 4, in <module>
    __import__('pkg_resources').run_script('twarc==1.0.6', 'twarc')
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 738, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1506, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/twarc-1.0.6-py2.7.egg/EGG-INFO/scripts/twarc", line 4, in <module>
    __import__('pkg_resources').run_script('twarc==1.0.6', 'twarc')
  File "build/bdist.linux-x86_64/egg/twarc.py", line 194, in main
    for thing in things:
  File "build/bdist.linux-x86_64/egg/twarc.py", line 795, in replies
    screen_name = tweet['user']['screen_name']
TypeError: string indices must be integers
```
The test.log lists an "hydrate" then an "archived errors" instead replies:
```
2017-03-01 16:37:25,261 INFO loading config /root/.twarc
2017-03-01 16:37:25,262 INFO creating http session
2017-03-01 16:37:25,262 INFO hydrating ['833504567644024832']
2017-03-01 16:37:25,406 INFO archived errors
```

**Comment by edsu** [comment]
When I run that I see this:

```
% twarc replies 833504567644024832
{"contributors": null, "truncated": false, "text": "RT @fed17fab: Buon compleanno #xm24!! 15 anni di autogestione, \u00e8 il vostro piano a non essere compatibile con #Bologna @PDBologna #XMresta\u2026", "is_quote_status": false, "in_reply_to_status_id": null, "id": 833504567644024832, "favorite_count": 0, "source": "<a href=\"http://twitter.com/download/iphone\" rel=\"nofollow\">Twitter for iPhone</a>", "retweeted": false, "coordinates": null, "entities": {"symbols": [], "user_mentions": [{"id": 1971407575, "indices": [3, 12], "id_str": "1971407575", "screen_name": "fed17fab", "name": "fedefab"}, {"id": 14070425, "indices": [119, 129], "id_str": "14070425", "screen_name": "PDBologna", "name": "PD Bologna"}], "hashtags": [{"indices": [30, 35], "text": "xm24"}, {"indices": [110, 118], "text": "Bologna"}, {"indices": [130, 138], "text": "XMresta"}], "urls": []}, "in_reply_to_screen_name": null, "in_reply_to_user_id": null, "retweet_count": 17, "id_str": "833504567644024832", "favorited": false, "retweeted_status": {"contributors": null, "truncated": true, "text": "Buon compleanno #xm24!! 15 anni di autogestione, \u00e8 il vostro piano a non essere compatibile con #Bologna @PDBologna\u2026 https://t.co/ivP7I1IHAN", "is_quote_status": false, "in_reply_to_status_id": null, "id": 833401087906422786, "favorite_count": 29, "source": "<a href=\"http://twitter.com/download/android\" rel=\"nofollow\">Twitter for Android</a>", "retweeted": false, "coordinates": null, "entities": {"symbols": [], "user_mentions": [{"id": 14070425, "indices": [105, 115], "id_str": "14070425", "screen_name": "PDBologna", "name": "PD Bologna"}], "hashtags": [{"indices": [16, 21], "text": "xm24"}, {"indices": [96, 104], "text": "Bologna"}], "urls": [{"url": "https://t.co/ivP7I1IHAN", "indices": [117, 140], "expanded_url": "https://twitter.com/i/web/status/833401087906422786", "display_url": "twitter.com/i/web/status/8\u2026"}]}, "in_reply_to_screen_name": null, "in_reply_to_user_id": null, "retweet_count": 17, "id_str": "833401087906422786", "favorited": false, "user": {"follow_request_sent": false, "has_extended_profile": false, "profile_use_background_image": false, "default_profile_image": false, "id": 1971407575, "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png", "verified": false, "translator_type": "none", "profile_text_color": "000000", "profile_image_url_https": "https://pbs.twimg.com/profile_images/695562638567333888/ETSPROC4_normal.png", "profile_sidebar_fill_color": "000000", "entities": {"description": {"urls": []}}, "followers_count": 196, "profile_sidebar_border_color": "000000", "id_str": "1971407575", "profile_background_color": "000000", "listed_count": 10, "is_translation_enabled": false, "utc_offset": 3600, "statuses_count": 1469, "description": "Astrofisico - on the rebel side", "friends_count": 381, "location": "Earth", "profile_link_color": "FAB81E", "profile_image_url": "http://pbs.twimg.com/profile_images/695562638567333888/ETSPROC4_normal.png", "following": false, "geo_enabled": false, "profile_banner_url": "https://pbs.twimg.com/profile_banners/1971407575/1454936355", "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png", "screen_name": "fed17fab", "lang": "en-gb", "profile_background_tile": false, "favourites_count": 670, "name": "fedefab", "notifications": false, "url": null, "created_at": "Sat Oct 19 11:52:30 +0000 2013", "contributors_enabled": false, "time_zone": "Rome", "protected": false, "default_profile": false, "is_translator": false}, "geo": null, "in_reply_to_user_id_str": null, "possibly_sensitive": false, "lang": "it", "created_at": "Sun Feb 19 19:41:17 +0000 2017", "in_reply_to_status_id_str": null, "place": null}, "user": {"follow_request_sent": false, "has_extended_profile": true, "profile_use_background_image": true, "default_profile_image": false, "id": 1377361808, "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png", "verified": false, "translator_type": "none", "profile_text_color": "333333", "profile_image_url_https": "https://pbs.twimg.com/profile_images/752137068474761217/-okU_tRr_normal.jpg", "profile_sidebar_fill_color": "DDEEF6", "entities": {"description": {"urls": []}}, "followers_count": 438, "profile_sidebar_border_color": "C0DEED", "id_str": "1377361808", "profile_background_color": "C0DEED", "listed_count": 11, "is_translation_enabled": false, "utc_offset": null, "statuses_count": 1523, "description": "matematica-femminista-tecnopolitica-gattini", "friends_count": 1222, "location": "", "profile_link_color": "1DA1F2", "profile_image_url": "http://pbs.twimg.com/profile_images/752137068474761217/-okU_tRr_normal.jpg", "following": false, "geo_enabled": false, "profile_banner_url": "https://pbs.twimg.com/profile_banners/1377361808/1468158627", "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png", "screen_name": "Sara_J27", "lang": "it", "profile_background_tile": false, "favourites_count": 1635, "name": "Pamela des Barres", "notifications": false, "url": null, "created_at": "Wed Apr 24 15:51:56 +0000 2013", "contributors_enabled": false, "time_zone": null, "protected": false, "default_profile": true, "is_translator": false}, "geo": null, "in_reply_to_user_id_str": null, "lang": "it", "created_at": "Mon Feb 20 02:32:29 +0000 2017", "in_reply_to_status_id_str": null, "place": null}
```

**Comment by remagio** [comment]
It is. And it's the unique instance on VM file system. Supposing package/files7version issues I cleaned up the VM at every test installing it via pip or src vi git clone:
```
/usr/local/bin/twarc
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info
/usr/local/lib/python2.7/dist-packages/twarc.py
/usr/local/lib/python2.7/dist-packages/twarc.pyc
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info/DESCRIPTION.rst
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info/INSTALLER
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info/METADATA
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info/RECORD
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info/WHEEL
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info/metadata.json
/usr/local/lib/python2.7/dist-packages/twarc-1.0.7.dist-info/top_level.txt
```

**Comment by edsu** [comment]
I can see in your error message pasted above that you are running v1.0.6; for example:

```
File "/usr/local/lib/python2.7/dist-packages/twarc-1.0.6-py2.7.egg/EGG-INFO/scripts/twarc", line 4, in <module>
    __import__('pkg_resources').run_script('twarc==1.0.6', 'twarc')
```

**Comment by remagio** [comment]
Got it, wrong paste, my fault b/c checking other VM/releases too including developments tree.
This is the above VM with 1.0.7 fresh installed giving with same ID:
```
user@store01:/tmp# twarc replies 833504567644024832

errors
Traceback (most recent call last):
  File "/usr/local/bin/twarc", line 4, in <module>
    twarc.main()
  File "/usr/local/lib/python2.7/dist-packages/twarc.py", line 194, in main
    for thing in things:
  File "/usr/local/lib/python2.7/dist-packages/twarc.py", line 795, in replies
    screen_name = tweet['user']['screen_name']
TypeError: string indices must be integers
```

**Comment by hugovk** [comment]
Working for me also, macOS Sierra, Python 2.7.12, twarc 
```bash
$ pip install -U twarc
Requirement already up-to-date: twarc in /usr/local/lib/python2.7/site-packages
Requirement already up-to-date: mock in /usr/local/lib/python2.7/site-packages (from twarc)
Requirement already up-to-date: unicodecsv in /usr/local/lib/python2.7/site-packages (from twarc)
Requirement already up-to-date: python-dateutil in /usr/local/lib/python2.7/site-packages (from twarc)
Collecting requests-oauthlib (from twarc)
  Downloading requests_oauthlib-0.8.0-py2.py3-none-any.whl
Requirement already up-to-date: pytest in /usr/local/lib/python2.7/site-packages (from twarc)
Requirement already up-to-date: funcsigs>=1; python_version < "3.3" in /usr/local/lib/python2.7/site-packages (from mock->twarc)
Collecting pbr>=0.11 (from mock->twarc)
  Downloading pbr-2.0.0-py2.py3-none-any.whl (98kB)
    100% |████████████████████████████████| 102kB 1.1MB/s 
Requirement already up-to-date: six>=1.9 in /usr/local/lib/python2.7/site-packages (from mock->twarc)
Requirement already up-to-date: requests>=2.0.0 in /usr/local/lib/python2.7/site-packages (from requests-oauthlib->twarc)
Requirement already up-to-date: oauthlib>=0.6.2 in /usr/local/lib/python2.7/site-packages (from requests-oauthlib->twarc)
Collecting setuptools (from pytest->twarc)
  Downloading setuptools-34.3.0-py2.py3-none-any.whl (389kB)
    100% |████████████████████████████████| 399kB 678kB/s 
Requirement already up-to-date: py>=1.4.29 in /usr/local/lib/python2.7/site-packages (from pytest->twarc)
Collecting appdirs>=1.4.0 (from setuptools->pytest->twarc)
  Downloading appdirs-1.4.2-py2.py3-none-any.whl
Requirement already up-to-date: packaging>=16.8 in /usr/local/lib/python2.7/site-packages (from setuptools->pytest->twarc)
Requirement already up-to-date: pyparsing in /usr/local/lib/python2.7/site-packages (from packaging>=16.8->setuptools->pytest->twarc)
Installing collected packages: requests-oauthlib, pbr, appdirs, setuptools
  Found existing installation: requests-oauthlib 0.7.0
    Uninstalling requests-oauthlib-0.7.0:
      Successfully uninstalled requests-oauthlib-0.7.0
  Found existing installation: pbr 1.10.0
    Uninstalling pbr-1.10.0:
      Successfully uninstalled pbr-1.10.0
  Found existing installation: appdirs 1.4.0
    Uninstalling appdirs-1.4.0:
      Successfully uninstalled appdirs-1.4.0
  Found existing installation: setuptools 34.2.0
    Uninstalling setuptools-34.2.0:
      Successfully uninstalled setuptools-34.2.0
Successfully installed appdirs-1.4.2 pbr-2.0.0 requests-oauthlib-0.8.0 setuptools-34.3.0
```
```bash
$ twarc replies 833504567644024832 --log test.log
{"contributors": null, "truncated": false, "text": "RT @fed17fab: Buon compleanno #xm24!! 15 anni di autogestione, \u00e8 il vostro piano a non essere compatibile con #Bologna @PDBologna #XMresta\u2026", "is_quote_status": false, "in_reply_to_status_id": null, "id": 833504567644024832, "favorite_count": 0, "source": "<a href=\"http://twitter.com/download/iphone\" rel=\"nofollow\">Twitter for iPhone</a>", "retweeted": false, "coordinates": null, "entities": {"symbols": [], "user_mentions": [{"id": 1971407575, "indices": [3, 12], "id_str": "1971407575", "screen_name": "fed17fab", "name": "fedefab"}, {"id": 14070425, "indices": [119, 129], "id_str": "14070425", "screen_name": "PDBologna", "name": "PD Bologna"}], "hashtags": [{"indices": [30, 35], "text": "xm24"}, {"indices": [110, 118], "text": "Bologna"}, {"indices": [130, 138], "text": "XMresta"}], "urls": []}, "in_reply_to_screen_name": null, "in_reply_to_user_id": null, "retweet_count": 17, "id_str": "833504567644024832", "favorited": false, "retweeted_status": {"contributors": null, "truncated": true, "text": "Buon compleanno #xm24!! 15 anni di autogestione, \u00e8 il vostro piano a non essere compatibile con #Bologna @PDBologna\u2026 https://t.co/ivP7I1IHAN", "is_quote_status": false, "in_reply_to_status_id": null, "id": 833401087906422786, "favorite_count": 29, "source": "<a href=\"http://twitter.com/download/android\" rel=\"nofollow\">Twitter for Android</a>", "retweeted": false, "coordinates": null, "entities": {"symbols": [], "user_mentions": [{"id": 14070425, "indices": [105, 115], "id_str": "14070425", "screen_name": "PDBologna", "name": "PD Bologna"}], "hashtags": [{"indices": [16, 21], "text": "xm24"}, {"indices": [96, 104], "text": "Bologna"}], "urls": [{"url": "https://t.co/ivP7I1IHAN", "indices": [117, 140], "expanded_url": "https://twitter.com/i/web/status/833401087906422786", "display_url": "twitter.com/i/web/status/8\u2026"}]}, "in_reply_to_screen_name": null, "in_reply_to_user_id": null, "retweet_count": 17, "id_str": "833401087906422786", "favorited": false, "user": {"follow_request_sent": false, "has_extended_profile": false, "profile_use_background_image": false, "default_profile_image": false, "id": 1971407575, "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png", "verified": false, "translator_type": "none", "profile_text_color": "000000", "profile_image_url_https": "https://pbs.twimg.com/profile_images/695562638567333888/ETSPROC4_normal.png", "profile_sidebar_fill_color": "000000", "entities": {"description": {"urls": []}}, "followers_count": 196, "profile_sidebar_border_color": "000000", "id_str": "1971407575", "profile_background_color": "000000", "listed_count": 10, "is_translation_enabled": false, "utc_offset": 3600, "statuses_count": 1469, "description": "Astrofisico - on the rebel side", "friends_count": 381, "location": "Earth", "profile_link_color": "FAB81E", "profile_image_url": "http://pbs.twimg.com/profile_images/695562638567333888/ETSPROC4_normal.png", "following": false, "geo_enabled": false, "profile_banner_url": "https://pbs.twimg.com/profile_banners/1971407575/1454936355", "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png", "screen_name": "fed17fab", "lang": "en-gb", "profile_background_tile": false, "favourites_count": 670, "name": "fedefab", "notifications": false, "url": null, "created_at": "Sat Oct 19 11:52:30 +0000 2013", "contributors_enabled": false, "time_zone": "Rome", "protected": false, "default_profile": false, "is_translator": false}, "geo": null, "in_reply_to_user_id_str": null, "possibly_sensitive": false, "lang": "it", "created_at": "Sun Feb 19 19:41:17 +0000 2017", "in_reply_to_status_id_str": null, "place": null}, "user": {"follow_request_sent": false, "has_extended_profile": true, "profile_use_background_image": true, "default_profile_image": false, "id": 1377361808, "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png", "verified": false, "translator_type": "none", "profile_text_color": "333333", "profile_image_url_https": "https://pbs.twimg.com/profile_images/752137068474761217/-okU_tRr_normal.jpg", "profile_sidebar_fill_color": "DDEEF6", "entities": {"description": {"urls": []}}, "followers_count": 438, "profile_sidebar_border_color": "C0DEED", "id_str": "1377361808", "profile_background_color": "C0DEED", "listed_count": 11, "is_translation_enabled": false, "utc_offset": null, "statuses_count": 1523, "description": "matematica-femminista-tecnopolitica-gattini", "friends_count": 1222, "location": "", "profile_link_color": "1DA1F2", "profile_image_url": "http://pbs.twimg.com/profile_images/752137068474761217/-okU_tRr_normal.jpg", "following": false, "geo_enabled": false, "profile_banner_url": "https://pbs.twimg.com/profile_banners/1377361808/1468158627", "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png", "screen_name": "Sara_J27", "lang": "it", "profile_background_tile": false, "favourites_count": 1635, "name": "Pamela des Barres", "notifications": false, "url": null, "created_at": "Wed Apr 24 15:51:56 +0000 2013", "contributors_enabled": false, "time_zone": null, "protected": false, "default_profile": true, "is_translator": false}, "geo": null, "in_reply_to_user_id_str": null, "lang": "it", "created_at": "Mon Feb 20 02:32:29 +0000 2017", "in_reply_to_status_id_str": null, "place": null}
```
---
I can see one possible problem: the exception comes from build/bdist.linux-x86_64/egg/twarc.py but you showed 1.0.7 installed at /usr/local/lib/python2.7/dist-packages/twarc.py. Why are these different?

Try this: uninstall twarc (`pip uninstall twarc -y`), and do it again (just in case, and repeat until there's nothing to uninstall). Then make sure `twarc` doesn't run; if it does, you need to manually remove it. Finally, install again (`pip install -U twarc` to also upgrade dependencies).

Failing that, find the twarc.py file, edit it to confirm the version number and as a test, replace:
```python
if __name__ == "__main__":
    main()
```
with:
```python
if __name__ == "__main__":
    print(123)
```
and rerun. If you don't see `123`, it's the wrong file. If you do, revert that, go to line 795 and add these prints and let us know what you get.
```python
        print(type(tweet))
        print(dir(tweet))
        print(tweet)
        screen_name = tweet['user']['screen_name']
```
I get:
```
<type 'dict'>
['__class__', '__cmp__', '__contains__', '__delattr__', '__delitem__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__iter__', '__le__', '__len__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setitem__', '__sizeof__', '__str__', '__subclasshook__', 'clear', 'copy', 'fromkeys', 'get', 'has_key', 'items', 'iteritems', 'iterkeys', 'itervalues', 'keys', 'pop', 'popitem', 'setdefault', 'update', 'values', 'viewitems', 'viewkeys', 'viewvalues']
{u'contributors': None, u'truncated': False, u'text': u'RT @fed17fab: Buon compleanno #xm24!! 15 anni di autogestione, \xe8 il vostro piano a non essere compatibile con #Bologna @PDBologna #XMresta\u2026', u'is_quote_status': False, u'in_reply_to_status_id': None, u'id': 833504567644024832, u'favorite_count': 0, u'source': u'<a href="http://twitter.com/download/iphone" rel="nofollow">Twitter for iPhone</a>', u'retweeted': False, u'coordinates': None, u'entities': {u'symbols': [], u'user_mentions': [{u'id': 1971407575, u'indices': [3, 12], u'id_str': u'1971407575', u'screen_name': u'fed17fab', u'name': u'fedefab'}, {u'id': 14070425, u'indices': [119, 129], u'id_str': u'14070425', u'screen_name': u'PDBologna', u'name': u'PD Bologna'}], u'hashtags': [{u'indices': [30, 35], u'text': u'xm24'}, {u'indices': [110, 118], u'text': u'Bologna'}, {u'indices': [130, 138], u'text': u'XMresta'}], u'urls': []}, u'in_reply_to_screen_name': None, u'in_reply_to_user_id': None, u'retweet_count': 17, u'id_str': u'833504567644024832', u'favorited': False, u'retweeted_status': {u'contributors': None, u'truncated': True, u'text': u'Buon compleanno #xm24!! 15 anni di autogestione, \xe8 il vostro piano a non essere compatibile con #Bologna @PDBologna\u2026 https://t.co/ivP7I1IHAN', u'is_quote_status': False, u'in_reply_to_status_id': None, u'id': 833401087906422786, u'favorite_count': 29, u'source': u'<a href="http://twitter.com/download/android" rel="nofollow">Twitter for Android</a>', u'retweeted': False, u'coordinates': None, u'entities': {u'symbols': [], u'user_mentions': [{u'id': 14070425, u'indices': [105, 115], u'id_str': u'14070425', u'screen_name': u'PDBologna', u'name': u'PD Bologna'}], u'hashtags': [{u'indices': [16, 21], u'text': u'xm24'}, {u'indices': [96, 104], u'text': u'Bologna'}], u'urls': [{u'url': u'https://t.co/ivP7I1IHAN', u'indices': [117, 140], u'expanded_url': u'https://twitter.com/i/web/status/833401087906422786', u'display_url': u'twitter.com/i/web/status/8\u2026'}]}, u'in_reply_to_screen_name': None, u'in_reply_to_user_id': None, u'retweet_count': 17, u'id_str': u'833401087906422786', u'favorited': False, u'user': {u'follow_request_sent': False, u'has_extended_profile': False, u'profile_use_background_image': False, u'default_profile_image': False, u'id': 1971407575, u'profile_background_image_url_https': u'https://abs.twimg.com/images/themes/theme1/bg.png', u'verified': False, u'translator_type': u'none', u'profile_text_color': u'000000', u'profile_image_url_https': u'https://pbs.twimg.com/profile_images/695562638567333888/ETSPROC4_normal.png', u'profile_sidebar_fill_color': u'000000', u'entities': {u'description': {u'urls': []}}, u'followers_count': 196, u'profile_sidebar_border_color': u'000000', u'id_str': u'1971407575', u'profile_background_color': u'000000', u'listed_count': 10, u'is_translation_enabled': False, u'utc_offset': 3600, u'statuses_count': 1469, u'description': u'Astrofisico - on the rebel side', u'friends_count': 381, u'location': u'Earth', u'profile_link_color': u'FAB81E', u'profile_image_url': u'http://pbs.twimg.com/profile_images/695562638567333888/ETSPROC4_normal.png', u'following': False, u'geo_enabled': False, u'profile_banner_url': u'https://pbs.twimg.com/profile_banners/1971407575/1454936355', u'profile_background_image_url': u'http://abs.twimg.com/images/themes/theme1/bg.png', u'screen_name': u'fed17fab', u'lang': u'en-gb', u'profile_background_tile': False, u'favourites_count': 670, u'name': u'fedefab', u'notifications': False, u'url': None, u'created_at': u'Sat Oct 19 11:52:30 +0000 2013', u'contributors_enabled': False, u'time_zone': u'Rome', u'protected': False, u'default_profile': False, u'is_translator': False}, u'geo': None, u'in_reply_to_user_id_str': None, u'possibly_sensitive': False, u'lang': u'it', u'created_at': u'Sun Feb 19 19:41:17 +0000 2017', u'in_reply_to_status_id_str': None, u'place': None}, u'user': {u'follow_request_sent': False, u'has_extended_profile': True, u'profile_use_background_image': True, u'default_profile_image': False, u'id': 1377361808, u'profile_background_image_url_https': u'https://abs.twimg.com/images/themes/theme1/bg.png', u'verified': False, u'translator_type': u'none', u'profile_text_color': u'333333', u'profile_image_url_https': u'https://pbs.twimg.com/profile_images/752137068474761217/-okU_tRr_normal.jpg', u'profile_sidebar_fill_color': u'DDEEF6', u'entities': {u'description': {u'urls': []}}, u'followers_count': 438, u'profile_sidebar_border_color': u'C0DEED', u'id_str': u'1377361808', u'profile_background_color': u'C0DEED', u'listed_count': 11, u'is_translation_enabled': False, u'utc_offset': None, u'statuses_count': 1523, u'description': u'matematica-femminista-tecnopolitica-gattini', u'friends_count': 1222, u'location': u'', u'profile_link_color': u'1DA1F2', u'profile_image_url': u'http://pbs.twimg.com/profile_images/752137068474761217/-okU_tRr_normal.jpg', u'following': False, u'geo_enabled': False, u'profile_banner_url': u'https://pbs.twimg.com/profile_banners/1377361808/1468158627', u'profile_background_image_url': u'http://abs.twimg.com/images/themes/theme1/bg.png', u'screen_name': u'Sara_J27', u'lang': u'it', u'profile_background_tile': False, u'favourites_count': 1635, u'name': u'Pamela des Barres', u'notifications': False, u'url': None, u'created_at': u'Wed Apr 24 15:51:56 +0000 2013', u'contributors_enabled': False, u'time_zone': None, u'protected': False, u'default_profile': True, u'is_translator': False}, u'geo': None, u'in_reply_to_user_id_str': None, u'lang': u'it', u'created_at': u'Mon Feb 20 02:32:29 +0000 2017', u'in_reply_to_status_id_str': None, u'place': None}
```

**Comment by remagio** [comment]
@edsu Till last night "pip install twarc --upgrade" install 1.0.7 but "python setup.py install" 1.0.6 (I have not compared src code but it still quote previous version string on docnow/twarc https://github.com/DocNow/twarc/blob/master/setup.py#L8). That's why previous traces looks different: one from VMs using pip and one from VMs using git clone's setup.

@hugovk VMs are cleaned before tests, verifying no related files of any type remains. Both kind (pip/git clone) are working since 1,5 year with no issues and constantly updated.

**Comment by remagio** [comment]
Hi @edsu,
I did some more tests and I found the source of the issue. Despite the presence of a populated .twarc profile, and previous twarc's jobs were still running fine, I started some new twarc with replies getting: "TypeError: string indices must be integers" using <twarc replies '833504567644024832'>".

TypeError confused me thinking it was related with "replies" but it related with Twitter Keys & config arguments. Despite previous releases -p argument stopped to work and default key profile too. Since last twarc updates (both VMs via pip or via github) it works only if settings ENVs Twitter Key or using --profile instead -p.
I missed twarc deprecated -p vs --profile (!?).

So the real issue is "if using <twarc replies '833504567644024832'> without
- a valid __--profile__ name (default not works)
- ENVs Twitter api keys
it says "TypeError: string indices must be integers" instead of something like "twitter api key or .twarc missing".
